### PR TITLE
[vs17.8] update arcade and fix build 

### PR DESCRIPTION
**Pull request by JanProvaznik**
### Context
https://github.com/dotnet/msbuild/pull/10808 cleanup
update dependencies from Arcade
fix build errors

### Changes Made

1. update dependencies from Arcade - enabled propagating updates through Darc .NET 8 servicing

2. remove BuildXL feed from nuget config - the feed has been deleted and the BuildXL packages are now taken from dotnet-tools feed.

3. Suppress warning IDE0305 failing https://github.com/dotnet/msbuild/runs/31503006515
```
src/Shared/CopyOnWriteDictionary.cs#L387

src/Shared/CopyOnWriteDictionary.cs(387,56): error IDE0305: (NETCORE_ENGINEERING_TELEMETRY=Build) Collection initialization can be simplified (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0305)
```

4. Suppress API compatibility errors failing https://github.com/dotnet/msbuild/runs/31553017667 , which appeared after taking the bugfixes to compat analyzer from arcade https://github.com/dotnet/msbuild/pull/10838#discussion_r1808368168

6. include PortableRuntimeIdentifierGraph.json whose absence fails here https://github.com/dotnet/msbuild/runs/31596126390

No functional/code changes.

### CODE REVIEW

**Review by YuliiaKovalova — APPROVED** [review]
@rainersigwald  can we send it to tactics as "tell mode"?

File: eng/dependabot/Packages.props [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <PackageVersion Include="LargeAddressAware" Version="1.0.5" />
     <PackageVersion Update="LargeAddressAware" Condition="'$(LargeAddressAwareVersion)' != ''" Version="$(LargeAddressAwareVersion)" />
 
-    <PackageVersion Include="Microsoft.BuildXL.Processes" Version="0.1.0-20230929.2" />
```
Comment (rainersigwald) [inline eng/dependabot/Packages.props:22]: I don't think we should take this in servicing. Can we mirror the older package instead?

File: src/Framework/CompatibilitySuppressions.xml [hunk 2]
```diff
@@ -1,7 +1,70 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <!-- PKV004 for netstandard2.0-supporting TFs that we do not have runtime assemblies for.
+  <Suppression>
+    <DiagnosticId>CP0008</DiagnosticId>
```
Comment (rainersigwald) [inline src/Framework/CompatibilitySuppressions.xml:5]: Is this just all stuff that popped up due to bugfixes in the compat analyzer?

**Comment by JanKrivanek** [comment]
Is it possible to list what fixes are we backporting here and why?

Security related fixes should definitely be backported. As for functional changes and even fixes - this might need some discussion.

File: src/Framework/CompatibilitySuppressions.xml (hunk 2 above)
Comment (YuliiaKovalova) [reply src/Framework/CompatibilitySuppressions.xml:5]: yes, exactly.

File: eng/dependabot/Packages.props (hunk 1 above)
Comment (JanProvaznik) [reply eng/dependabot/Packages.props:22]: This version of the package came from a deleted BuildXL feed.
I looked into it and what currently happens on the branch is that neither of these versions are in any available feed so it restores the latest Microsoft.BuildXL.Processes 0.1.0-20240606.3 from https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json 

Who can I contact to add the old BuildXL 0.1.0-20230929.2 to the feed? I don't know where it comes from and who's responsible for the feed. Alternative solution: we could update this branch to point to 20240606.3

Also I noticed that Microsoft.BuildXL.Processes is restored from https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json while Microsoft.BuildXL.Utilities.Core and Microsoft.BuildXL.Native from https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json
shouldn't they both come from the same feed?

Comment (rainersigwald) [reply eng/dependabot/Packages.props:22]: @JanProvaznik I'll add you to an internal thread on this.

**Comment by JanProvaznik** [comment]
Will revert [0932640](https://github.com/dotnet/msbuild/pull/10838/commits/0932640338f8216f7acba318b4fce31720e084ef) once Microsoft.BuildXL.Processes is in the new feed and then there will be no changes to the code, only the build and warning suppressions.

**Review by rainersigwald — APPROVED** [review]

**Pull request merged**